### PR TITLE
Music add play validations

### PR DESCRIPTION
**Pull request by breville**
This adds two new validations: `played_chords` (aka "play notes") and `played_patterns` (aka "play drums").  

They work the same as the existing `played_sounds` validation.  

The existing `played_sounds`, `played_sounds_together`, `played_sound_triggered`, `played_sound_id` validations now works specifically for "play" blocks (and not for "play notes" or "play drums" blocks).

### CODE REVIEW

**Review by thomasoniii — APPROVED** [review]
lgtm 🚀

File: apps/src/music/progress/MusicValidator.ts [hunk 1]
```diff
@@ -47,62 +49,97 @@ export default class MusicValidator extends Validator {
...
 
-        this.conditionsChecker.addSatisfiedCondition({
-          name: MusicConditions.PLAYED_SOUND_ID.name,
-          value: eventData.id,
-        });
-      }
-
-      if (eventData.when <= currentPlayheadPosition) {
-        playedNumberSounds++;
+        if (eventData.when <= currentPlayheadPosition) {
```
Comment (sanchitmalhotra126) [inline apps/src/music/progress/MusicValidator.ts]: Could we just check `eventData.when <= currentPlayheadPosition` once outside of these nested ifs, and early return if false?

File: apps/src/music/progress/MusicValidator.ts [hunk 2]
```diff
@@ -47,62 +49,97 @@ export default class MusicValidator extends Validator {
...
-        });
-      }
+      this.conditionsChecker.addSatisfiedCondition({
+        name: MusicConditions.PLAYED_SOUNDS.name,
+        value: numberSounds,
+      });
+    }
+
+    // Check for up to a certain number of patterns played.
+    for (
```
Comment (sanchitmalhotra126) [inline apps/src/music/progress/MusicValidator.ts]: worth creating a small helper for these? something like `addPlayedConditions(conditionName: string, playedNumber: number)`?

File: apps/src/music/progress/MusicValidator.ts (hunk 1 above)
Comment (breville) [reply apps/src/music/progress/MusicValidator.ts]: Good idea; done.

File: apps/src/music/progress/MusicValidator.ts (hunk 2 above)
Comment (breville) [reply apps/src/music/progress/MusicValidator.ts]: Good idea; done.

**Review by sanchitmalhotra126 — APPROVED** [review]

**Pull request merged**